### PR TITLE
Refactor GitHub Actions for infra deployment

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR refactors the GH actions to reuse the steps required to prepare/execute Terraform commands. 
The apply/plan workflows now use a composite action which encapsulates all the steps needed to execute/summarize terraform commands.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204610208832639/1204691860848733/f) _(internal)_

## 🔍 What does this change?

- Refactors github actions for TF

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

## 🐾 Next steps

- Use this flow in other projects we have that need CI/CD terraform jobs

## 🛡 What tests cover this?

- CI

## ❓ How to test this?

You can execute the workflows manually with dispatches, i've executed both here:
https://github.com/hashintel/hash/actions/runs/5176563531?pr=2639 (plan)
https://github.com/hashintel/hash/actions/runs/5176564761?pr=2639  (apply)

### CODE REVIEW

File: .github/actions/terraform-exec/action.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      working-directory: ${{ inputs.working-directory }}
+      shell: bash
+      run: terraform init -no-color 2>&1
+
+    - name: Terraform Validate
+      id: validate
+      shell: bash
+      run: terraform validate -no-color 2>&1
+
+    - name: "[${{ inputs.env }}] Terraform Plan"
```
Comment (TimDiekmann) [inline .github/actions/terraform-exec/action.yml:64]:
```suggestion
    - name: "[${{ inputs.env }}] Terraform ${{ inputs.command }}"
```

Comment (thehabbos007) [reply .github/actions/terraform-exec/action.yml:64]: Thanks! My search auto-enables case sensitivity and i miss these :(

File: .github/actions/terraform-exec/action.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        aws-access-key-id: ${{ steps.secrets.outputs.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ steps.secrets.outputs.AWS_SECRET_ACCESS_KEY }}
+        aws-session-token: ${{ steps.secrets.outputs.AWS_SESSION_TOKEN }}
+        aws-region: us-east-1
+
+    - name: "[${{ inputs.env }}] Terraform Init"
```
Comment (TimDiekmann) [inline .github/actions/terraform-exec/action.yml:51]: Given that `plan` or `apply` is lowercase, I'd change this for consistency as well
```suggestion
    - name: "[${{ inputs.env }}] Terraform init"
```

Comment (thehabbos007) [reply .github/actions/terraform-exec/action.yml:51]: The plan/apply will be lower-case with `${{ inputs.cmd }}` though

Comment (TimDiekmann) [reply .github/actions/terraform-exec/action.yml:51]: Yes, this is why I suggested to make this title lowercase as well 🙂

Comment (thehabbos007) [reply .github/actions/terraform-exec/action.yml:51]: Oh sorry i saw the diff as the opposite. I'm blind xd

Comment (TimDiekmann) [reply .github/actions/terraform-exec/action.yml]: I wish I could say that never happens to me 😂

**Review by TimDiekmann — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**